_posts/2016/04/2016-04-20-tips-tricks-live-coding.md

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+authors: ["Lex Nederbragt"]
+title: "10 tips and tricks for instructing and teaching by means of live coding"
+date: 2016-04-20
+time: "00:00:00"
+category: ["education", "instructor-training"]
+---
+
+One of the key teaching practices used at Software and Data Carpentry workshops is 'live coding': instructors don't use slides, but work through the lesson material, typing in the code or instructions, with the workshop participants following along. Learning how to teach using live-coding is best done in practice, with feedback from peers (this is why it is included in [instructor training](http://swcarpentry.github.io/instructor-training/08-practices.html). Nonetheless, this post lists ten tips and tricks to help you get started.
+
+This text (re)uses, and expands on, elements from the Software and Data Carpentry [instructor training](http://swcarpentry.github.io/instructor-training/08-practices.html) materials.
+
+
+### 1. Be seen and heard
+
+If you are physically able to stand up for a couple of hours, do it while you are teaching. When you sit down, you are hiding yourself behind others for those sitting in the back rows. Make sure to notify the workshop organisers of your wish to stand up and, ask them to arrange a high table/standing desk or [lectern](https://en.wikipedia.org/wiki/Lectern#Academic_context).
+
+Regardless of whether you are standing or sitting, make sure to move around occasionally. You can for example move to the screen to point something out, or draw something on the white/blackboard (see below). Moving around makes the teaching more lively, less monotonous. It draws the learners' attention away from their screens, to you, which helps getting the point you are making across.
+
+Even though you may have a good voice and know how to use it well, it may be an idea to use a microphone, especially if the workshop room is equipped with one. Your voice will be less tired, and you increase the chance of people with hearing difficulties being able to follow the workshop.
+
+### 2. Take it slow
+
+For every command you type, every word of code you write, every menu item or website button you click, say out loud what you are doing while you do it. This not only slows you down, it allows learners who are following along to copy what you do, even when they are looking at their screen while doing it. If the output of your command or code makes what you just typed disappear from view, scroll back up so learners can see it again - this is especially needed for the unix shell lesson.
+
+Other possibilities are to execute the same command a second time, or copy and paste the last command(s) into the workshop Etherpad.
```

Comment:
These days, I:

1. Run the command (speaking it as I go).
2. Turn and point to the command and its output on the screen and go through it a second time.

This gives slow typists a chance to catch up, and people who may not have heard the first time a chance to check their understanding.